src/ripple/app/tx/impl/SetOracle.cpp

```diff
@@ -194,6 +197,15 @@ adjustOwnerCount(ApplyContext& ctx, int count)
     return false;
 }
 
+static void
+setPriceDataInnerObjTemplate(STObject& obj)
+{
+    if (SOTemplate const* elements =
+            InnerObjectFormats::getInstance().findSOTemplateBySField(
+                sfPriceData))
+        obj.set(*elements);
+}
+
```

Comment:
What is this doing? What is a serialized object template?

Reply:
Object template sets the fields style - required, optional, default. As it turns out (see #4906), the inner object template has to be set explicitly, otherwise the default field doesn't work as expected. I.e., accessing this field results in an error. `Oracle` object has an inner object `PriceData`, which contains a default field `sfScale`. This is why this code is added.